lightningd/peer_control.c

```diff
@@ -1339,6 +1343,101 @@ static const struct json_command getinfo_command = {
 };
 AUTODATA(json_command, &getinfo_command);
 
+
+static struct command_result *json_setchannelfee(struct command *cmd,
+					 const char *buffer,
+					 const jsmntok_t *obj UNNEEDED,
+					 const jsmntok_t *params)
+{
+	struct json_stream *response;
+	const jsmntok_t *idtok;
+	struct peer *peer;
+	struct channel *channel;
+	struct channel_id cid;
+	struct command_result *res;
+	struct amount_msat *base, base_default;
+	u32 *ppm, ppm_default;
+
+	/* Use daemon config values base/ppm as default */
+	amount_msat_from_u32(&base_default, cmd->ld->config.fee_base);
+	ppm_default = cmd->ld->config.fee_per_satoshi;
+
+	/* Parse the JSON command */
+	if (!param(cmd, buffer, params,
+			p_req("id", param_tok, &idtok),
+			p_opt_def("base", param_msat, &base, base_default),
+			p_opt_def("ppm", param_number, &ppm, ppm_default),
+			NULL))
+		return command_param_failed();
+
+	/* Check base for u32 overflow */
+	if (amount_msat_overflow_u32(*base))
+		return command_fail(cmd, JSONRPC2_INVALID_PARAMS,
+				"Value 'base' exceeds u32 max");
+
+	/* Find the channel */
+	peer = peer_from_json(cmd->ld, buffer, idtok);
+	if (peer) {
+		channel = peer_active_channel(peer);
+		if (!channel)
+			return command_fail(cmd, LIGHTNINGD,
+					"Could not find active channel of peer with that id");
+	} else {
+		res = command_find_channel(cmd, buffer, idtok, &channel, &peer);
+		if (res)
+			return res;
+		/* check channel is found and in valid state */
+		if (!channel)
+			return command_fail(cmd, LIGHTNINGD,
+					"Could not find channel with that id");
+	}
+
+	if (channel->state != CHANNELD_NORMAL)
+		return command_fail(cmd, LIGHTNINGD,
+				"Channel is not in state CHANNELD_NORMAL");
```

Comment:
as a nicety, i'd include the channel id and current state in the fail message.

Reply:
addressed by 
773bc94